concordium-node/src/consensus_ffi/ffi.rs

```diff
@@ -1596,6 +1643,57 @@ impl ConsensusContainer {
         Ok((out_hash, out_data))
     }
 
+    /// Get the entire smart contract state of the specified instance.
+    pub fn get_instance_state_v2(
+        &self,
+        block_hash: &crate::grpc2::types::BlockHashInput,
+        address: &crate::grpc2::types::ContractAddress,
+    ) -> Result<([u8; 32], ContractStateResponse), tonic::Status> {
+        use crate::grpc2::Require;
+        let (block_id_type, block_hash) =
+            crate::grpc2::types::block_hash_input_to_ffi(block_hash).require()?;
+        let addr_index = address.index;
+        let addr_subindex = address.subindex;
+        let consensus = self.consensus.load(Ordering::SeqCst);
+        let mut out_v0_data: Vec<u8> = Vec::new();
+        let mut out_v1_data = V1ContractStateReceiver {
+            state:  None,
+            loader: None,
+        };
```

Comment:
Would it makes sense to lift the Option to the entire struct? 
It would remove some of the cases when matching.
If not, maybe add a comment on why at the definition of the struct.

Reply:
That also works. Changed it.